completers/nixos-rebuild_completer/cmd/root.go

```diff
@@ -0,0 +1,269 @@
+package cmd
+
+import (
+	"github.com/carapace-sh/carapace"
+	"github.com/carapace-sh/carapace-bin/pkg/actions/net"
+	"github.com/carapace-sh/carapace-bin/pkg/actions/tools/nix"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "nixos-rebuild",
+	Short: "reconfigure a NixOS machine",
+	Long:  "https://nixos.org",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+func Execute() error {
+	return rootCmd.Execute()
+}
+
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	rootCmd.PersistentFlags().Bool("help", false, "Show usage information")
+	rootCmd.PersistentFlags().Bool("offline", false, "Disable substituters and consider all previously downloaded files up-to-date")
+	rootCmd.PersistentFlags().StringSlice("option", []string{}, "Set the Nix configuration setting name to value")
+	rootCmd.PersistentFlags().Bool("refresh", false, "Consider all previously downloaded files out-of-date")
+	rootCmd.PersistentFlags().Bool("version", false, "Show version information")
```

Comment:
I think I've got the wrong version of `nixos-rebuild`?
It doesn't have some flags like `--version` and `--access-tokens`.
It seems to accept the others on root level as well (PersistentFlags) :thinking: .

Reply:
What version do you have?

Reply:
The ones that gets installed from `23.11` and `unstable`.

Reply:
It seems you're right - the man page said the flags needed to come after the subcommand, so I didn't test them before, but everything (except `--json` on `list-generations`) seems to actually be PersistentFlags. The nixos-rebuild man page also claims it accepts the flags from the `nix` command, but apparently it's only a subset

Reply:
Testing further, it seems the command accepts flags beyond the list the specifically call out in the man page (for example nixos-rebuild accepts the `--offline` flag from the nix command), but it isn't consistent in that it accepts "Logging-related options" or other option groups, the accepted flags seems to be taken arbitrarily

Reply:
https://github.com/NixOS/nixpkgs/blob/nixos-23.11/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh#L48

Reply:
I've updated the PR to make the flags persistent and to only accept flags found in that script